flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/event/FileSystemJobEventStore.java

```diff
@@ -0,0 +1,365 @@
+        }
+
+        // get read files.
+        try {
+            readIndex = 0;
+            readFiles = getAllJobEventPaths();
+        } catch (IOException e) {
+            throw new IOException("Cannot init filesystem job event store.", e);
+        }
+        // set write index
+        writeIndex = readFiles.size();
+
+        // create job event write executor.
+        this.eventWriterExecutor = createJobEventWriterExecutor();
+        eventWriterExecutor.scheduleAtFixedRate(
+                () -> {
+                    if (outputStream != null && !corrupted) {
+                        try {
+                            outputStream.flush();
+                        } catch (Exception e) {
+                            LOG.warn(
+                                    "Flush event file {} meet error, will not record event any more.",
+                                    writeFile,
+                                    e);
+                            corrupted = true;
+                            closeOutputStream();
+                        }
+                    }
+                },
+                0L,
+                flushIntervalInMs,
+                TimeUnit.MILLISECONDS);
+
+        corrupted = false;
+    }
+
+    /** Get all the paths of event files available in workingDir. */
+    private List<Path> getAllJobEventPaths() throws IOException {
+        List<Path> paths = new ArrayList<>();
+        int index = INITIAL_INDEX;
+        Path path = new Path(workingDir, FILE_PREFIX + index++);
+        while (fileSystem.exists(path)) {
+            paths.add(path);
+            path = new Path(workingDir, FILE_PREFIX + index++);
+        }
+        return paths;
+    }
+
+    protected ScheduledExecutorService createJobEventWriterExecutor() {
+        return Executors.newSingleThreadScheduledExecutor(
+                new ExecutorThreadFactory("job-event-writer"));
+    }
+
+    @Override
+    public void stop(boolean clear) {
+        try {
+            // close output stream
+            eventWriterExecutor.execute(this::closeOutputStream);
+
+            // close input stream
+            closeInputStream();
+
+            // close writer executor.
+            if (eventWriterExecutor != null) {
+                eventWriterExecutor.shutdown();
+                try {
+                    if (!eventWriterExecutor.awaitTermination(10, TimeUnit.SECONDS)) {
+                        eventWriterExecutor.shutdownNow();
+                    }
+                } catch (InterruptedException ignored) {
+                    eventWriterExecutor.shutdownNow();
+                }
+                eventWriterExecutor = null;
+            }
+
+            if (clear) {
+                fileSystem.delete(workingDir, true);
+            }
+        } catch (Exception exception) {
+            LOG.warn("Fail to stop filesystem job event store.", exception);
+        }
+    }
+
+    void writeInt(FSDataOutputStream out, int num) throws IOException {
```

Comment:
This method can be part of `FsBatchFlushOutputStream`.
`FsBatchFlushOutputStream` can be explicit used in this store.